File: apps/mobile/ios/DeviceUtilsIOS.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(DeviceUtilsIOS, NSObject)

RCT_EXTERN_METHOD(isTablet:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)

@end

File: apps/mobile/ios/DeviceUtilsIOS.swift
import UIKit

@objc(DeviceUtilsIOS)
class DeviceUtilsIOS: NSObject {
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
  
  @objc
  func isTablet(_ resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
    resolve(UIDevice.current.userInterfaceIdiom == .pad)
  }
  
  @objc
  func constantsToExport() -> [String: Any]! {
    return [
      "isTablet": UIDevice.current.userInterfaceIdiom == .pad
    ]
  }
}
